developers need a way to test the gitbook build locally

/kind feature

**Describe the solution you'd like**

I would like to be able to test the gitbook build on my local system.

I tried using `hack/build-gitbook.sh` and encountered several errors, likely because that script is assuming it is running in CI.

```
$ bash -x ./hack/build-gitbook.sh
+ set -o errexit
+ set -o nounset
+ set -o pipefail
++ dirname ./hack/build-gitbook.sh
+ export KUBE_ROOT=./hack/..
+ KUBE_ROOT=./hack/..
+ cd ./hack/..
+ pushd docs/book/
~/go/src/github.com/kubernetes-sigs/cluster-api/docs/book ~/go/src/github.com/kubernetes-sigs/cluster-api
+ npm install gitbook-cli -g
npm WARN checkPermissions Missing write access to /usr/lib/node_modules
npm ERR! path /usr/lib/node_modules
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall access
npm ERR! Error: EACCES: permission denied, access '/usr/lib/node_modules'
npm ERR!  { [Error: EACCES: permission denied, access '/usr/lib/node_modules']
npm ERR!   stack:
npm ERR!    'Error: EACCES: permission denied, access \'/usr/lib/node_modules\'',
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/usr/lib/node_modules' }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator (though this is not recommended).
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dhellmann/.npm/_logs/2019-06-10T22_26_14_413Z-debug.log
```

I then removed the `-g` option from the `npm install` command and made more progress, but still failed because the bin directory for the local npm install was not in my $PATH.

```
$ bash -x ./hack/build-gitbook.sh
+ set -o errexit
+ set -o nounset
+ set -o pipefail
++ dirname ./hack/build-gitbook.sh
+ export KUBE_ROOT=./hack/..
+ KUBE_ROOT=./hack/..
+ cd ./hack/..
+ pushd docs/book/
~/go/src/github.com/kubernetes-sigs/cluster-api/docs/book ~/go/src/github.com/kubernetes-sigs/cluster-api
+ npm install gitbook-cli

> phantomjs-prebuilt@2.1.16 install /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt
> node install.js

PhantomJS not found on PATH
Downloading https://github.com/Medium/phantomjs/releases/download/v2.1.1/phantomjs-2.1.1-linux-x86_64.tar.bz2
Saving to /tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
Receiving...
  [======================================--] 95%
Received 22866K total.
Extracting tar contents (via spawned process)
Removing /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt/lib/phantom
Copying extracted folder /tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2-extract-1560205695242/phantomjs-2.1.1-linux-x86_64 -> /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt/lib/phantom
Writing location.js file
Done. Phantomjs binary available at /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs
+ gitbook-cli@2.3.2
added 718 packages from 786 contributors and audited 3483 packages in 22.87s
found 93 vulnerabilities (6 low, 59 moderate, 28 high)
  run `npm audit fix` to fix them, or `npm audit` for details
+ npm install phantomjs-prebuilt

> phantomjs-prebuilt@2.1.16 install /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt
> node install.js

PhantomJS not found on PATH
Download already available at /tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
Verified checksum of previously downloaded file
Extracting tar contents (via spawned process)
Removing /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt/lib/phantom
Copying extracted folder /tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2-extract-1560205704133/phantomjs-2.1.1-linux-x86_64 -> /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt/lib/phantom
Writing location.js file
Done. Phantomjs binary available at /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs
+ phantomjs-prebuilt@2.1.16
updated 2 packages and audited 3483 packages in 8.184s
found 93 vulnerabilities (6 low, 59 moderate, 28 high)
  run `npm audit fix` to fix them, or `npm audit` for details
+ npm ci
npm WARN prepare removing existing node_modules/ before installation

> phantomjs-prebuilt@2.1.16 install /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt
> node install.js

PhantomJS not found on PATH
Download already available at /tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
Verified checksum of previously downloaded file
Extracting tar contents (via spawned process)
Removing /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt/lib/phantom
Copying extracted folder /tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2-extract-1560205713186/phantomjs-2.1.1-linux-x86_64 -> /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt/lib/phantom
Writing location.js file
Done. Phantomjs binary available at /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs
added 161 packages in 8.051s
+ gitbook install
./hack/build-gitbook.sh: line 29: gitbook: command not found
```

So, I ran the gitbook install command by hand and it was successful. Then I ran `gitbook build` by hand and it failed.

```
$ ./node_modules/.bin/gitbook build
info: 13 plugins are installed
info: 12 explicitly listed
info: loading plugin "panel"... OK
info: loading plugin "sequence-diagrams"... OK
info: loading plugin "ga"... OK
info: loading plugin "codegroup"... OK
info: loading plugin "include-codeblock"...
warn: ace features disabled (`gitbook-plugin-ace` required)
OK
info: loading plugin "highlight"... OK
info: loading plugin "search"... OK
info: loading plugin "lunr"... OK
info: loading plugin "sharing"... OK
info: loading plugin "fontsettings"... OK
info: loading plugin "theme-api"... OK
info: loading plugin "theme-default"... OK
info: found 18 pages
info: found 27 asset files
warn: "options" property is deprecated, use config.get(key) instead

Error: ENOENT: no such file or directory, stat '/home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/_book/gitbook/gitbook-plugin-codegroup/code.group.css'
```

However, the missing file does exist:

```
$ ls -al /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/_book/gitbook/gitbook-plugin-codegroup/code.group.css
-rw-rw-r--. 1 dhellmann dhellmann 0 Jun 10 18:34 /home/dhellmann/go/src/github.com/kubernetes-sigs/cluster-api/docs/book/_book/gitbook/gitbook-plugin-codegroup/code.group.css
```

**Anything else you would like to add:**

There are lots of ways to solve this, and I don't really care which we use. We could add a flag to the existing script, wrap it in a container, or write a whole new script.

--- a/docs/book/RELEASE
+++ b/docs/book/RELEASE
@@ -12,7 +12,18 @@ content is generated into [./docs/book/_book](../../docs/book/_book) but not
 checked into version control. Netlify will automatically rebuild the 
 documentation on each merge to master.
 
+## Submit a PR
+
+## Testing with Netlify
+
+Each PR to the cluster-api repository causes mutliple CI stages to be run. One
+of the stages is named `deploy/netlify` and by clicking on the `Details` link
+from the PR you can browse the generated content before it is deployed live.
+
 ## Testing locally
+
+This has only been verified to work on OSX. For platform agnostic testing see
+the [Testing with Netlify](#testing-with-netlify) section above.
 
 ```
 hack/build-gitbook.sh
@@ -29,8 +40,6 @@ cd docs/book
 gitbook serve
 # open localhost:4000 in a web browser
 ```
-
-## Finally submit a PR
 
 [markdown]: https://toolchain.gitbook.com/syntax/markdown.html
 [gitbook]: https://toolchain.gitbook.com/

